src/main/java/com/es/phoneshop/service/ProductService.java

```diff
@@ -1,17 +1,22 @@
 package com.es.phoneshop.service;
 
+import com.es.phoneshop.SortField;
+import com.es.phoneshop.SortOrder;
 import com.es.phoneshop.model.entity.Product;
 
 import java.util.List;
 
 public interface ProductService {
+
     Product getProductById(Long id);
 
     List<Product> getProducts();
 
-    void deleteProduct(Long id);
+    List<Product> findProductsByQuery(String query);
 
-    void changeState(Product product, boolean state);
+    List<Product> getProductsWithSortingAndQuery(SortField sortField, SortOrder sortOrder, String query);
```

Comment:
method namings should be consistent:  getProducts...() and findProducts...() should have single prefix